models/appModels.js

```diff
@@ -8,26 +8,74 @@ const fetchTopics = () => {
 }
 
 
-const fetchArticles = () => {
-    return db.query('SELECT article_id FROM comments;').then(({rows}) => {
+const fetchArticles = (order = 'desc', sort_by = 'created_at', topic) => {
+    const greenlistedSortBys = ['created_at', 'article_id', 'title', 'created_at', 'votes']
+    const greenlistedOrders = ['asc', 'desc']
+
+  
+let queryString = `SELECT * FROM articles `
+
+if (topic !== undefined){
+    queryString += 'WHERE topic = $1 '
+    if (!greenlistedSortBys.includes(sort_by) || !greenlistedOrders.includes(order)){
+        return Promise.reject({status:400, msg: 'Bad Request'})
+    } else {
+
+        queryString += `ORDER BY ${sort_by} ${order}`
+
+    
+        return db.query(queryString, [topic]).then(({rows}) => {
+    const topicArticles = rows
+
+   return db.query(`SELECT article_id FROM comments;`).then(({rows}) => {
```

Comment:
This might be better achieved using a an agregate function (COUNT()) in SQL rather than using map. We generally want to reserve sorting and counting data for the database rather than doing it in out model.